scripts/modules/data/models/behaviors.mjs

```diff
@@ -0,0 +1,119 @@
+import {MODULE} from "../../constants.mjs";
+
+const {BooleanField, HTMLField, NumberField, SchemaField, StringField} = foundry.data.fields;
+
+/**
+ * Initialization function to register data models.
+ */
+export default function initializeBehaviors() {
+  CONFIG.RegionBehavior.dataModels[`${MODULE.ID}.trap`] = TrapData;
+  CONFIG.RegionBehavior.typeIcons[`${MODULE.ID}.trap`] = "fa-solid fa-person-falling-burst";
+}
+
+/**
+ * Behavior type that configures a trap that prompts a saving throw and deals damage.
+ */
+class TrapData extends foundry.data.regionBehaviors.RegionBehaviorType {
+  /** @override */
+  static events = {
+    [CONST.REGION_EVENTS.TOKEN_ENTER]: foundry.utils.debounce(TrapData.#onTokenMoveIn, 150)
+  };
+
+  /* -------------------------------------------------- */
+
+  /** @override */
+  static LOCALIZATION_PREFIXES = ["MYTHACRI.REGION.TRAP"];
```

Comment:
Regarding localization, this is the array that tells Foundry where to find the strings.